lib/identity_config_reader.rb

```diff
@@ -0,0 +1,47 @@
+class IdentityConfigReader
+  attr_accessor :root_path
+
+  def initialize(
+    root_path: Rails.root,
+    s3_client: nil,
+    logger: Logger.new(STDOUT)
+  )
+    @root_path = root_path
+    @logger = logger
+    @s3_client = s3_client
+  end
+
+  def read_configuration(write_copy_to: nil)
+    configuration = read_default_configuration.deep_merge(
+      read_override_configuration,
+    )
+
+    if write_copy_to && !File.exists?(write_copy_to)
```

Comment:
😬 can we pull this into the gem? or is it urgent to get this merged before that?

Reply:
not stopping until we have `identity-hostdata@999999.99999`

Reply:
Hostdata PR: https://github.com/18F/identity-hostdata/pull/26